### ISSUE TITLE
Class comments replication

### ISSUE THREAD

**Issue by VaclavPokorny** [immutables/immutables#476]
I have a project where I use generated types everywhere. I use a utility (http://enunciate.webcohesion.com/) which generates documentation for REST API from those generated classes. This works nicely for descriptions of getters which are copied from abstract type and the Enunciate is able to use them. However, the description of the class itself is replaced with something like this:

```
/**
 * Immutable implementation of {@link AbstractTestData}.
 * <p>
 * Use the builder to create immutable instances:
 * {@code TestData.builder()}.
 * Use the static factory method to create immutable instances:
 * {@code new TestData()}.
 */
```

I would like to have a to have a style option simliar to `headerComments` which would force replication of abstract type's class comment to it's generated type(s). I would name it something like `classComments`.

**Comment by elucash** [comment]
Makes sense. How about we will go configuration-less. When using inverse generation (like immutable Val extends hand-written AbstractVal), we'll try to extract javadoc from a abstract value type and put it instead of the first paragraph like this:

```java
/**
 * COPY OF CLASS JAVADOC THAT WAS ORIGINALLY PRESENT
 * <p>
 * Use the builder to create immutable instances:
 * {@code TestData.builder()}.
 * Use the static factory method to create immutable instances:
 * {@code new TestData()}.
 */
```

**Comment by VaclavPokorny** [comment]
Well, in my case this would make generated REST API documentation contain it's implementation details. But it is still good enough for me.

**Comment by elucash** [comment]
How about just use javadoc if present, without additions after `<p>`

/**
 * COPY OF CLASS JAVADOC THAT WAS ORIGINALLY PRESENT
 */

**Comment by VaclavPokorny** [comment]
Would be ok.

**Comment by elucash** [comment]
available in v2.3.8 (currently propagating to Central). Copies Javadoc if available, no configuration, just depends on generated class to be public (vs package-private Abstract one) or have simple name (`typeImmutable="*"`)
